## Programming for Data Analysis - Project 2

#### Martin Cusack        

#### Student ID: G00239124


### Introduction

In this project I will conduct some analysis and investigation of a data set sourced from the UC Irvine Machine Learning Repository, containing data on Breast Cancer samples gathered by a team at the University of Wisconsin Hospitals and Clinics. I hope to train a number machine learning algorithms using the data and synthesise data etc... I will also conduct a literature review

In [1]:
# import Python libraries
import numpy as np
import pandas as pd
import seaborn as sns
import sklearn.neighbors as nei

In [2]:
# import Breast Cancer Wisconsin (Original) Data Set into a pandas data frame
df=pd.read_csv (r"C:\Users\cusac\OneDrive\Desktop\breast_cancer_bd.csv")
df

Sample code number  Clump Thickness  Uniformity of Cell Size  \
0               1000025                5                        1   
1               1002945                5                        4   
2               1015425                3                        1   
3               1016277                6                        8   
4               1017023                4                        1   
..                  ...              ...                      ...   
694              776715                3                        1   
695              841769                2                        1   
696              888820                5                       10   
697              897471                4                        8   
698              897471                4                        8   

     Uniformity of Cell Shape  Marginal Adhesion  Single Epithelial Cell Size  \
0                           1                  1                            2   
1                           4                  5                            7   
2                           1                  1                            2   
3                           8                  1                            3   
4                           1                  3                            2   
..                        ...                ...                          ...   
694                         1                  1                            3   
695                         1                  1                            2   
696                        10                  3                            7   
697                         6                  4                            3   
698                         8                  5                            4   

    Bare Nuclei  Bland Chromatin  Normal Nucleoli  Mitoses  Class  
0             1                3                1        1      2  
1            10                3                2        1      2  
2             2                3                1        1      2  
3             4                3                7        1      2  
4             1                3                1        1      2  
..          ...              ...              ...      ...    ...  
694           2                1                1        1      2  
695           1                1                1        1      2  
696           3                8               10        2      4  
697           4               10                6        1      4  
698           5               10                4        1      4  

[699 rows x 11 columns]

In [10]:
# display number of rows and columns
df.shape

(699, 11)

In [9]:
# show count of attributes in"Class"
print(df['Class'].value_counts())

2    458
4    241
Name: Class, dtype: int64


### Breast Cancer Wisconsin (Original) Data Set

The data set which I will be investigating for this project was created at the University of Wisconsin Hospitals in Madison, Wisconsin, USA, by a team led by Dr. William Wolberg.  It is a multi-variate data set which is available in two different forms, **Original** and **Diagnostic**.  For the purposes of this project I chose to investigate the Original version.

As shown above the Original Data Set has 699 rows and 11 columns.  The first column, Sample code number, is primarily used for identification purposes and displays a 6 or 7 digit sample number.  The next 9 columns ("**Clump Thickness", "Uniformity of Cell Size", "Uniformity of Cell Shape", "Marginal Adhesion", "Single Epithelial Cell Size", "Bare Nuclei", "Bland Chromatin", "Normal Nucleoli", "Mitoses")** all refer to different measurements of breast cancer tissue samples. The final column ("Class") seperates the samples into either Class 2 or Class 4.  As shown above using the .value_counts() function, the data set includes 458 instances of class 2 and 241 instances of class 4.